metrics.go

```diff
@@ -1,19 +1,47 @@
 package f3
 
 import (
+	"context"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/metric"
 )
 
 var meter = otel.Meter("f3")
 var metrics = struct {
-	headDiverged       metric.Int64Counter
-	reconfigured       metric.Int64Counter
-	manifests_received metric.Int64Counter
+	headDiverged      metric.Int64Counter
+	reconfigured      metric.Int64Counter
+	manifestsReceived metric.Int64Counter
+	validationTime    metric.Int64Histogram
 }{
-	headDiverged:       must(meter.Int64Counter("f3_head_diverged", metric.WithDescription("Number of times we encountered the head has diverged from base scenario."))),
-	reconfigured:       must(meter.Int64Counter("f3_reconfigured", metric.WithDescription("Number of times we reconfigured due to new manifest being delivered."))),
-	manifests_received: must(meter.Int64Counter("f3_manifests_received", metric.WithDescription("Number of manifests we have received"))),
+	headDiverged:      must(meter.Int64Counter("f3_head_diverged", metric.WithDescription("Number of times we encountered the head has diverged from base scenario."))),
+	reconfigured:      must(meter.Int64Counter("f3_reconfigured", metric.WithDescription("Number of times we reconfigured due to new manifest being delivered."))),
+	manifestsReceived: must(meter.Int64Counter("f3_manifests_received", metric.WithDescription("Number of manifests we have received"))),
+	validationTime: must(meter.Int64Histogram("f3_validation_time_ms",
+		metric.WithDescription("Histogram of time spent validating broadcasted in milliseconds"),
+		metric.WithExplicitBucketBoundaries(5.0, 10.0, 20.0, 30.0, 40.0, 50.0, 100.0, 200.0, 300.0, 400.0, 500.0, 1000.0),
```

Comment:
Does the Histogram give us the `_sum` metric as well?

Reply:
I would also add maybe two buckets of smaller sizes (I'm being optimistic).

Reply:
> Does the Histogram give us the `_sum` metric as well?

It does yes.

> I'm being optimistic

I like it!